### PR TITLE
Add new gmf.source.Swisstopo

### LINKED ISSUE #328: Add new gmf.source.Swisstopo
To access tiles from swisstopo, see https://api3.geo.admin.ch/services/sdiservices.html#wmts

### PR DESCRIPTION
**Pull request by fredj**
work in progress
Fixes #328

### CODE REVIEW

**Comment by elemoine** [comment]
Is this now ready for review?

**Comment by fredj** [comment]
not yet: I want to test it in a project before.

Note that this source has no example because of the access restriction

**Comment by elemoine** [comment]
Some docs about the access restrictions would be nice.

File: contribs/gmf/externs/gmfx.js [hunk 1]
```diff
@@ -142,3 +142,34 @@ gmfx.ServiceUrls;
...
+ * @type {string}
+ */
+gmfx.source.SwisstopoOptions.prototype.format;
+
+
+/**
+ * The `Time` dimension of the source.
+ * @type {string}
+ */
+gmfx.source.SwisstopoOptions.prototype.timestamp;
```
Comment (pgiraud) [inline contribs/gmf/externs/gmfx.js]: Could this be put just after the AsitVDOptions source?

Comment (fredj) [inline contribs/gmf/externs/gmfx.js]: done

File: contribs/gmf/src/source/swisstoposource.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ *          registered by Swisstopo to use the service.
+ * @see https://api3.geo.admin.ch/services/sdiservices.html#wmts
+ *
+ * @constructor
+ * @extends {ol.source.WMTS}
+ * @param {gmfx.source.SwisstopoOptions} options WMTS options.
+ */
+gmf.source.Swisstopo = function(options) {
+
+  var format = options.format ? options.format : 'png';
```
Comment (pgiraud) [inline contribs/gmf/src/source/swisstoposource.js]: You're declaring the `format` variable but not using it afterwards.

Comment (fredj) [inline contribs/gmf/src/source/swisstoposource.js]: fixed

File: contribs/gmf/src/source/swisstoposource.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+  goog.base(this, {
+    attributions: [gmf.source.Swisstopo.ATTRIBUTION_],
+    url: 'https://wmts{5-9}.geo.admin.ch/1.0.0/{Layer}/default/{Time}' +
+        '/21781/{TileMatrix}/{TileRow}/{TileCol}.' + options.format,
```
Comment (pgiraud) [inline contribs/gmf/src/source/swisstoposource.js]: `format` instead?

Comment (fredj) [inline contribs/gmf/src/source/swisstoposource.js]: fixed

**Comment by sbrunner** [comment]
Thanks @pgiraud , @fredj, for me it can me merged :-)

**Comment by fredj** [comment]
Thanks for the review @pgiraud

**Pull request merged**